workspaces/ui-v2/src/components/sharing/ShareButton.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogActions,
+  makeStyles,
+  Typography,
+} from '@material-ui/core';
+import ShareIcon from '@material-ui/icons/Share';
+import { ShareModal } from './ShareModal';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useSpecRepository } from '<src>/contexts/SpecRepositoryContext';
+import { useAppConfig } from '<src>/contexts/config/AppConfiguration';
+import { useAsyncMemo } from 'use-async-memo';
+
+export enum ShareTarget {
+  TEAM,
+  CUSTOMER,
```

Comment:
I believe this is meant to be Consumer? As in, if I am sharing my API I either want to discuss/share it with my team or with external consumers? But there's overlap in the venn diagram between teammates and consumers. So perhaps this can be revisited.